llvm/lib/CodeGen/GlobalISel/Utils.cpp

```diff
@@ -1745,18 +1778,55 @@ static bool canCreateUndefOrPoison(Register Reg, const MachineRegisterInfo &MRI,
                                    UndefPoisonKind Kind) {
   MachineInstr *RegDef = MRI.getVRegDef(Reg);
 
-  if (auto *GMI = dyn_cast<GenericMachineInstr>(RegDef)) {
-    if (ConsiderFlagsAndMetadata && includesPoison(Kind) &&
-        GMI->hasPoisonGeneratingFlags())
-      return true;
-  } else {
-    // Conservatively return true.
-    return true;
-  }
```

Comment:
Are regular `MachineInstr`s not capable of generating undef or poison?

Reply:
ValueTracking writes it as follows:
```
if (ConsiderFlagsAndMetadata && includesPoison(Kind) &&
      Op->hasPoisonGeneratingAnnotations())
    return true;
```
The question is how to handle non `GenericMachineInstr`.

Reply:
The IRTranslator only attaches flags to Gisel Instructions aka `GenericMachineInstr`. The test is only about poison generating flags resp. annotations. Sure an AARCH64 div instruction can generate poison by division by 0, but it has no poison generating flags.